src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPPollingSource.java

```diff
@@ -87,4 +122,159 @@ public void onTerminate() {
     endUsingFolder();
   }
 
+  @Override
+  protected Message[] getMessages(Folder openFolder) {
+    if (!getPredicateBuilder().isPresent() || !getPredicateBuilder().get().getEnableRemoteSearchFilter()) {
```

Comment:
The remote search filter feature should also be applied to the imap list operation.

Reply:
Looks like we are using pagination to fetch results for list operation. Unfortunately java mail does not support applying both, pagination and server side filtering. I did, however, move the "remoteSeachFilterEnabled" flag to the imap-listener element, because given this limitation it would not make sense to have it in the matcher element.